lib/parseaddr.c

```diff
@@ -314,7 +314,7 @@ static int parseaddr_domain(char **inp, char **domainp, char **commentp, int *in
 
     for (;;) {
         c = *src++;
-        if (Uisalnum(c) || c == '-' || c == '[' || c == ']' || c == ':') {
+        if (Uisalnum(c) || c == '-' || c == '[' || c == ']' || c == ':' || c > 127) {
             *dst++ = c;
```

Comment:
Do we care to make sure that the char is in the (0x80 <= c <= 0xF4)  range?

Reply:
I'd say not to. Unless we change the parseaddr to check if the byte sequence is truly valid UTF-8, we should leave that to the client.